twitter/src/main/java/com/hazelcast/jet/contrib/twitter/TwitterSources.java

```diff
@@ -0,0 +1,394 @@
+     * <p>
+     * Example usage:
+     * <pre>{@code
+     * List<Long> userIds = new ArrayList<Long>(
+     *              Arrays.asList(612473L, 759251L, 1367531L, 34713362L, 51241574L));
+     * StreamSource<String> timestampedStreamSource =
+     *              TwitterSources.timestampedStream(
+     *                      credentials,
+     *                      () -> new StatusesFilterEndpoint().followings(userIds),
+     *              );
+     * Pipeline p = Pipeline.create();
+     * StreamStage<String> srcStage = p.readFrom(timestampedStreamSource);
+     * }</pre>
+     *
+     * @param credentials      a Twitter OAuth1 credentials that consists
+     *                         "consumerKey", "consumerSecret", "token",
+     *                         "tokenSecret" keys.
+     * @param endpointSupplier Supplier that supplies a Twitter
+     *                         StreamingEndpoint to connect to source.
+     * @return a timestamped stream source to use in
+     * {@link com.hazelcast.jet.pipeline.Pipeline#readFrom}
+     */
+    @Nonnull
+    public static StreamSource<String> timestampedStream(
+            @Nonnull Properties credentials,
+            @Nonnull SupplierEx<? extends StreamingEndpoint> endpointSupplier
+    ) {
+        return timestampedStream(credentials, Constants.STREAM_HOST, endpointSupplier);
+    }
+
+    /**
+     * The method {@link TwitterSources#timestampedStream(Properties, SupplierEx)}
+     * is overloaded here. This method differs from the other by enabling the
+     * users to choose a different Twitter Streaming API host. e.g.
+     * {@link Constants#ENTERPRISE_STREAM_HOST}
+     * <p>
+     * Example usage:
+     * <pre>{@code
+     * List<Long> userIds = new ArrayList<Long>(
+     *              Arrays.asList(612473L, 759251L, 1367531L, 34713362L, 51241574L));
+     * StreamSource<String> timestampedStreamSource =
+     *              TwitterSources.timestampedStream(
+     *                      credentials,
+     *                      Constants.STREAM_HOST,
+     *                      () -> new StatusesFilterEndpoint().followings(userIds),
+     *              );
+     * Pipeline p = Pipeline.create();
+     * StreamStage<String> srcStage = p.readFrom(timestampedStreamSource);
+     * }</pre>
+     *
+     * @param credentials      a Twitter OAuth1 credentials that consists
+     *                         "consumerKey", "consumerSecret", "token",
+     *                         "tokenSecret" keys.
+     * @param host             a Twitter host URL to connect. These hosts are
+     *                         defined in {@link com.twitter.hbc.core.Constants}.
+     * @param endpointSupplier Supplier that supplies a Twitter
+     *                         StreamingEndpoint to connect to source.
+     * @return a timestamped stream source to use in
+     * {@link com.hazelcast.jet.pipeline.Pipeline#readFrom}
+     */
+
+    @Nonnull
+    public static StreamSource<String> timestampedStream(
```

Comment:
Do we need both `stream` and `timestampedStream`? We can have just the timestamped variant and the user might use the native timestamps or add his own, if needed.

Reply:
The reason might be to avoid parsing the message if we don't need the timestamp. Maybe we should document that:
>If you don't use timestamps, prefer to use {@link #stream} to avoid the overhead of parsing the message.

Reply:
The only purpose for that is to avoid parsing as you state. I will add your statement to javadoc.